src/components/ContractorPayment/EditModal/EditModalPresentation.tsx

```diff
@@ -0,0 +1,152 @@
+import { useState } from 'react'
+import type { ContractorDataStrict } from '../types'
+import { useComponentContext } from '@/contexts/ComponentAdapter/useComponentContext'
+import { Flex, Grid, ActionsLayout } from '@/components/Common'
+import { formatNumberAsCurrency } from '@/helpers/formattedStrings'
+import { useLocale } from '@/contexts/LocaleProvider/useLocale'
+
+type ContractorData = ContractorDataStrict
+
+interface ContractorPaymentEditModalProps {
+  contractor: ContractorData
+  onSave: (contractor: ContractorData) => void
+  onCancel: () => void
+}
+
+export const ContractorPaymentEditModal = ({
+  contractor,
+  onSave,
+  onCancel,
+}: ContractorPaymentEditModalProps) => {
+  const { Button, Text, Heading, Card, RadioGroup, NumberInput } = useComponentContext()
+  const { locale } = useLocale()
+
+  const [editedContractor, setEditedContractor] = useState<ContractorData>({
+    ...contractor,
+  })
+
+  const calculateTotal = () => {
+    const wageAmount =
+      editedContractor.wageType === 'Hourly'
+        ? editedContractor.hours * (editedContractor.hourlyRate || 0)
+        : editedContractor.wage
+    return wageAmount + editedContractor.bonus + editedContractor.reimbursement
+  }
+
+  const handleFieldChange = (field: keyof ContractorData, value: number | string) => {
+    setEditedContractor(prev => ({
+      ...prev,
+      [field]: value,
+      total: calculateTotal(),
+    }))
+  }
```

Comment:
Thinking this could be omitted as forms can be wired up in a subsequent commit?

Form here will also need to be updated to use `Field` components and hook form. not sure if that's within scope here, might be preferable to just keep the current form components as placeholders and hook form can be wired up in a follow up?